plugins/modules/android_sdk.py

```diff
@@ -0,0 +1,201 @@
+    description: Detailed information of how to install and use sdkmanager command line tool.
+    link: https://developer.android.com/tools/sdkmanager
+'''
+
+EXAMPLES = r'''
+- name: Before installing Android SDK packages, all licenses must be accepted
+  shell: "yes | sdkmanager --licenses"
+
+- name: Install build-tools;34.0.0
+  community.general.android_sdk:
+    name: build-tools;34.0.0
+    state: present
+
+- name: Install build-tools;34.0.0 and platform-tools
+  community.general.android_sdk:
+    name:
+      - build-tools;34.0.0
+      - platform-tools
+    state: present
+
+- name: Delete build-tools;34.0.0
+  community.general.android_sdk:
+    name: build-tools;34.0.0
+    state: absent
+
+- name: Install platform-tools or update if installed
+  community.general.android_sdk:
+    name: platform-tools
+    state: latest
+
+- name: Install build-tools;34.0.0 to a different SDK root
+  community.general.android_sdk:
+    name: build-tools;34.0.0
+    state: present
+    sdk_root: "/path/to/new/root"
+
+- name: Install a package from another channel
+  community.general.android_sdk:
+    name: some-package-present-in-canary-channel
+    state: present
+    channel: canary
+'''
+
+RETURN = r'''
+installed:
+    description: a list of packages that have been installed
+    returned: when packages have changed
+    type: list
+    sample: ['build-tools;34.0.0', 'platform-tools']
+
+removed:
+    description: a list of packages that have been removed
+    returned: when packages have changed
+    type: list
+    sample: ['build-tools;34.0.0', 'platform-tools']
+'''
+
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.sdkmanager import sdkmanager_runner, Package, \
+    AndroidSdkManager
+
+
+class AndroidSdk(StateModuleHelper):
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=['present', 'absent', 'latest']),
+            package=dict(type='list', elements='str', aliases=['pkg', 'name']),
+            sdk_root=dict(type='path'),
+            channel=dict(type='str', default='stable', choices=['stable', 'beta', 'dev', 'canary'])
+        ),
+        supports_check_mode=True
+    )
+    use_old_vardict = False
+    output_params = ('installed', 'removed')
+    change_params = ('installed', 'removed')
+
+    def __init_module__(self):
+        self.sdkmanager = AndroidSdkManager(sdkmanager_runner(self.module))
+        self.vars.set('installed', [], change=True)
+        self.vars.set('removed', [], change=True)
+
+    def _parse_packages(self):
+        arg_pkgs = self.vars.package
+        packages = set()
+        for arg_pkg in arg_pkgs:
+            package = Package(arg_pkg)
+            packages.add(package)
+
+        if len(packages) < len(arg_pkgs):
+            self.do_raise("Packages may not repeat")
+        return packages
```

Comment:
By looking at the code, `Package` is basically a wrapper for `name: str`. That being said, it might be faster to validate before creating the objects with:
```suggestion
        arg_pkgs = set(self.vars.package)
        if len(arg_pkgs) < len(self.vars.package):
            self.do_raise("Packages may not repeat")
        return set([Package(p) for p in arg_pkg])
```
(and using a comprehension to make it more concise - that part is totally up to you, I like comprehensions, some people hate them)

Reply:
done